### PR TITLE
Update emacs-*-devel versions, fix variants and misc clean ups

### PR DESCRIPTION
**Pull request by wyuenho**
#### Description

emacs|emas-devel|emacs-app|emacs-app-devel: bump revisions and fix variants

* set emacs-devel and emacs-app-devel to use HEAD
* move dbus variant under emacs and emacs-devel because it makes no sense to enable that when building the NS port app subports
* hide inline variant from emacs-app-devel
* only patch src/dbusbind.c when building dbus variant
* update git.url
* bump revisions for emacs and emacs-app
* remove emacs-devel src/dbusbind.c patch cos	it's the same as emacs'

References:
https://trac.macports.org/ticket/56423
https://github.com/macports/macports-ports/pull/1936
https://github.com/macports/macports-ports/pull/1948

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.5
Xcode 9.4

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by wyuenho** [comment]
@drkp @l2dy how does this look?

File: editors/emacs/Portfile [hunk 1]
```diff
@@ -52,18 +52,10 @@ if {$subport eq $name || $subport eq "emacs-app"} {
 }
 
 if {$subport eq "emacs-devel" || $subport eq "emacs-app-devel"} {
-    version         20170918
-    revision        1
+    version         HEAD
```
Comment (l2dy) [inline editors/emacs/Portfile]: IIRC, we don't use HEAD as version.

Comment (wyuenho) [reply editors/emacs/Portfile]: what do you use when a number doesn't make sense?

**Comment by drkp** [comment]
Ugh, I wasn't paying close enough attention; I shouldn't have merged that!

**Comment by wyuenho** [comment]
What's the Macports convention for recompile-from-HEAD-everytime-I-force-reinstall then?

**Comment by drkp** [comment]
There is none.

**Comment by wyuenho** [comment]
What's the point of a "devel" port if it has to be "reproducible"? The point of such ports is exactly because people want the bleeding edge, so if they want to update a devel port, they can just do a force reinstall. This isn't exactly a new concept, lots of linux package managers have something like this. Homebrew has dev channels, using HEAD is the closest to such concepts short of building out an entire infrastructure to support that.

You are going to have to deal with tickets regardless of whether a devel port is pinned to a git revision or not. Its unstable nature by definition means lots of maintenance, if you set it to head, people can just forget about it and force a reinstall every time he wants a new build from HEAD.

Also, I've never heard of a emacs revision that fails at building, so I don't think we should exaggerate its impact.

**Comment by l2dy** [comment]
@wyuenho e.g. if one of our patches fails to apply... This could happen.

**Comment by drkp** [comment]
Fixed up in 8747064.

**Comment by wyuenho** [comment]
There are no patches for the devel ports by default. The dbus variant for emac-devel and mutilfont_color patch for emacs-app-devel are both one-liners, so highly unlikely they'll break. I've already hidden the inline patch from emacs-app-devel.

Anyway if you pin to a revision, might as well remove the devel ports and tell uses to go compile it yourselves...

**Comment by jmroot** [comment]
If you really want to, you can specify an arbitrary git.branch on the command line when installing.

**Comment by wyuenho** [comment]
Is this yet another undocumented feature?

**Comment by jmroot** [comment]
It's documented, just not recommended for common use.

Packaging is a form of curation. If you're getting a non-deterministic version of the software when you install, then as you say you might as well remove the devel ports and tell users to go compile it themselves.

**Comment by ryandesign** [comment]
> Also, the buildbot has at least in the past had trouble with git:// urls, which is why we were using http fetch. I am going to change that back too.

When MacPorts was hosted at Apple, we had to contend with the very restrictive settings of Apple's corporate firewall. Now that we are hosting elsewhere, those restrictions no longer affect our buildbot. However, they may still affect users who install from source from restrictive networks, so using http or https is probably the most compatible.

**Pull request merged**